.github/workflows/repackage.yml

```diff
@@ -0,0 +1,193 @@
+    runs-on: ubuntu-latest
+    env:
+      BUILD_DATA_DIR: .build-data
+    outputs:
+      parallel-jobs-number: ${{ steps.split-packages.outputs.parallel-jobs-number }}
+      parallel-array: ${{ steps.split-packages.outputs.parallel-array }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: google-github-actions/auth@v1
+        with:
+          credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_KERNEL_CACHE }}'
+
+      - uses: google-github-actions/setup-gcloud@v1
+      
+      - uses: ./.github/actions/env
+
+      - name: Prepare Cache
+        run: |
+          mkdir -p "${BUILD_DATA_DIR}/cache"
+          IFS=',' read -r -a bucket_names <<< "${KERNEL_BUNDLE_BUCKET}"
+          for bucket in "${bucket_names[@]}"; do
+            if [ ! -f "${BUILD_DATA_DIR}/cache/cache.yml" ]; then
+              gsutil cp "${bucket}/cache.yml" "${BUILD_DATA_DIR}/cache/cache.yml" || true
+            fi
+          done
+          touch "${BUILD_DATA_DIR}/cache/cache.yml"
+
+      - name: Restore manifest
+        uses: actions/download-artifact@v3
+        with:
+          # this is archived by the crawl workflow
+          # see .github/workflows/crawl.yml for details
+          name: kernel-crawler-manifest
+
+      - name: List packages
+        run: make list-files
+
+      - name: Split package file
+        id: split-packages
+        run: |
+          import math
+          import os
+          import json
+
+          packages = []
+          with open('.build-data/packages.txt') as pkg:
+            packages = list(map(str.strip, pkg.readlines()))
+
+          parallel_jobs = math.ceil(len(packages) / 500)
+          if parallel_jobs > 32:
+            parallel_jobs = 32
+
+          # split the packages into chunks
+          jobs = [packages[i::parallel_jobs] for i in range(parallel_jobs)]
+          parallel = list(range(parallel_jobs))
+
+          combined = {id: jobs[id] for id in range(parallel_jobs)}
+
+          with open(os.environ['GITHUB_OUTPUT'], 'a') as output:
+            output.write(f'parallel-jobs-number={parallel_jobs}\n')
+            output.write(f'parallel-array={json.dumps(parallel)}\n')
+
+          with open('jobs.json', 'w') as j:
+            j.write(json.dumps(combined))
+
+        shell: python
+
+      # This is primarily for sanity checking
+      - name: Archive package list
+        uses: actions/upload-artifact@v3
+        with:
+          name: kernel-repackage-list
+          path: |
+            .build-data/packages.txt
+
+      - name: Archive Jobs and Cache
+        uses: actions/upload-artifact@v3
+        with:
+          name: package-jobs
+          path: |
+            .build-data/cache/cache.yml
+            jobs.json
+
+  repackage:
+    runs-on: ubuntu-latest
+    needs: split-tasks
+    if: ${{ needs.split-tasks.outputs.parallel-jobs-number > 0 }}
+    strategy:
+      fail-fast: false
+      matrix:
+        packer: ${{ fromJSON(needs.split-tasks.outputs.parallel-array) }}
```

Comment:
From what I understood, `parallel-array` is simply a list of parallel jobs `[0..parallel_job]`.  Later on this parameter is being used as an index into the jobs map. Does it mean GHA will pick up one number from this array somehow?

Reply:
yeah GHA will set `packer` to one of the numbers from the array and run it in parallel with all the others. The values themselves are somewhat meaningless and are just used to trigger an appropriate number of parallel runs